src/magicbox/SuggestionsManager.ts

```diff
@@ -141,73 +143,17 @@ export class SuggestionsManager {
     this.keyboardFocusedElement = null;
   }
 
-  public mergeSuggestions(suggestions: Array<Promise<Suggestion[]> | Suggestion[]>, callback?: (suggestions: Suggestion[]) => void) {
-    let results: Suggestion[] = [];
-    let timeout;
-    let stillNeedToResolve = true;
-    // clean empty / null values in the array of suggestions
-    suggestions = compact(suggestions);
-    const promise = (this.pendingSuggestion = new Promise<Suggestion[]>((resolve, reject) => {
-      // Concat all promises results together in one flat array.
-      // If one promise take too long to resolve, simply skip it
-      each(suggestions, (suggestion: Promise<Suggestion[]>) => {
-        let shouldRejectPart = false;
-        setTimeout(function() {
-          shouldRejectPart = true;
-          stillNeedToResolve = false;
-        }, this.options.timeout);
-        suggestion.then((item: Suggestion[]) => {
-          if (!shouldRejectPart && item) {
-            results = results.concat(item);
-          }
-        });
-      });
-
-      // Resolve the promise when one of those conditions is met first :
-      // - All suggestions resolved
-      // - Timeout is reached before all promises have processed -> resolve with what we have so far
-      // - No suggestions given (length 0 or undefined)
-      const onResolve = () => {
-        if (stillNeedToResolve) {
-          if (timeout) {
-            clearTimeout(timeout);
-          }
-          if (results.length == 0) {
-            resolve([]);
-          } else if (promise == this.pendingSuggestion || this.pendingSuggestion == null) {
-            resolve(results.sort((a, b) => b.index - a.index));
-          } else {
-            reject('new request queued');
-          }
-        }
-        stillNeedToResolve = false;
-      };
-
-      if (suggestions.length == 0) {
-        onResolve();
-      }
-      if (suggestions == undefined) {
-        onResolve();
-      }
+  public async receiveSuggestions(suggestions: Array<Promise<Suggestion[]> | Suggestion[]>) {
```

Comment:
For array types, we usually use the bracket syntax. e.g. `Promise<Suggestion[]>[]`

Also, if Suggestion is an interface, we usually add an `I` in front. e.g. `ISuggestion`. This refactor could be in a separate PR though.

Reply:
Good point, converting it from `Array<Promise<Suggestion[]> | Suggestion[]>` to `(Promise<Suggestion[]> | Suggestion[])[]`. I kinda wish Typescript also had symbol for promises.

As for `ISuggestion`, I was thinking of converting it in [JSUI-2700](https://coveord.atlassian.net/browse/JSUI-2700).